cache-config/t3cutil/toreq/conversions.go

```diff
@@ -40,3 +48,93 @@ func dsesToLatest(dses []tc.DeliveryServiceV40) []atscfg.DeliveryService {
 func jobsToLatest(jobs []tc.InvalidationJobV4) []atscfg.InvalidationJob {
 	return atscfg.ToInvalidationJobs(jobs)
 }
+
+// GetJobsCompat gets jobs from any Traffic Ops built from the ATC `master` branch, and converts the different formats to the latest.
+// This makes t3c work with old or new Traffic Ops deployed from `master`,
+// though it doesn't make a version of t3c older than this work with a new TO,
+// which isn't logically possible from the client.
+func (cl *TOClient) GetJobsCompat(opts toclient.RequestOptions) (tc.InvalidationJobsResponseV4, toclientlib.ReqInf, error) {
+	path := "/jobs"
+
+	objs := struct {
+		Response []InvalidationJobV4PlusLegacy `json:"response"`
+		tc.Alerts
+	}{}
+
+	if len(opts.QueryParameters) > 0 {
+		path += "?" + opts.QueryParameters.Encode()
+	}
+	reqInf, err := cl.c.TOClient.Req(http.MethodGet, path, nil, opts.Header, &objs)
+	if err != nil {
+		return tc.InvalidationJobsResponseV4{}, reqInf, errors.New("request: " + err.Error())
+	}
+
+	resp := tc.InvalidationJobsResponseV4{Alerts: objs.Alerts}
+	for _, job := range objs.Response {
+		newJob, err := InvalidationJobV4FromLegacy(job) // (InvalidationJobV4, error) {
+		if err != nil {
+			return tc.InvalidationJobsResponseV4{}, reqInf, errors.New("converting job from possible legacy format: " + err.Error())
+		}
+		resp.Response = append(resp.Response, newJob)
+	}
+	return resp, reqInf, nil
+}
+
+// InvalidationJobV4ForLegacy is a type alias to prevent MarshalJSON recursion.
+type InvalidationJobV4ForLegacy tc.InvalidationJobV4
+
+// InvalidationJobV4PlusLegacy has the data to deserialize both the latest and older versions that Traffic Ops could return.
+type InvalidationJobV4PlusLegacy struct {
+	StartTime *string `json:"startTime"`
```

Comment:
Just for my own edification, even though `InvalidationJobV4ForLegacy` contains a `StartTime` field with the same json tag, the JSON unmarshaller will unmarshal the `"startTime"` value into _this_ field because it's top-level rather than in an embedded field? If so, it would be good to comment something along those lines for future reference.

Reply:
Yes. Added.